hack/crdbversions/main.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2021 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v2"
+)
+
+// List of templates and destinations
+var targets = []struct{ template, output string }{
+	{"config/templates/operator.yaml.in", "manifests/operator.yaml"},
+	{"config/templates/deployment_patch.yaml.in", "manifests/patches/deployment_patch.yaml"},
+	{"config/templates/crdb-tls-example.yaml.in", "config/samples/crdb-tls-example.yaml"},
+	{"config/templates/example.yaml.in", "examples/example.yaml"},
+}
+
+// crdb-versions.yaml structure
+type crdbVersions struct {
+	CrdbVersions []string `yaml:"CrdbVersions"`
+}
+
+type templateData struct {
+	CrdbVersions            []*semver.Version
+	LatestStableCrdbVersion string
+	OperatorVersion         string
+	GeneratedWarning        string
+}
+
+// readCrdbVersions reads CRDB versions from a YAML file and sorts them
+// according to the semantic version sorting rules
+func readCrdbVersions(r io.Reader) ([]*semver.Version, error) {
+	contents, err := io.ReadAll(r)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open CRDB version file: %w", err)
+	}
+	var versions crdbVersions
+	if err := yaml.Unmarshal(contents, &versions); err != nil {
+		return nil, fmt.Errorf("cannot parse CRDB version file: %w", err)
+	}
+	var vs []*semver.Version
+	for _, raw := range versions.CrdbVersions {
+		v, err := semver.NewVersion(raw)
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert version `%s`: %w", r, err)
+		}
+		vs = append(vs, v)
+	}
+	sort.Sort(semver.Collection(vs))
+	return vs, nil
```

Comment:
Do we test that we have at least some versions?  What happens if we have an empty file?

Reply:
It's happening later, when we check if we have at least 1 stable version in the file. It's not a universal approach, but works fine for our case, where we have many stable versions.